File: src/main/java/com/team10/backend/global/exception/BusinessException.java
package com.team10.backend.global.exception;

import org.springframework.http.HttpStatus;

public class BusinessException extends RuntimeException {

    private final ErrorCode errorCode;

    public BusinessException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }

    public BusinessException(ErrorCode errorCode, String detail) {
        super(detail); // errorCode.message 대신 상세 메시지 사용
        this.errorCode = errorCode;
    }

    public HttpStatus getStatus() {
        return errorCode.getStatus();
    }

    public ErrorCode getErrorCode() {
        return this.errorCode;
    }
}

File: src/main/kotlin/com/team10/backend/global/exception/BusinessException.kt
package com.team10.backend.global.exception

import org.springframework.http.HttpStatus

open class BusinessException : RuntimeException {
    val errorCode: ErrorCode

    constructor(errorCode: ErrorCode) : super(errorCode.message) {
        this.errorCode = errorCode
    }

    constructor(errorCode: ErrorCode, detail: String?) : super(detail) // errorCode.message 대신 상세 메시지 사용
    {
        this.errorCode = errorCode
    }

    protected constructor(errorCode: ErrorCode, message: String, cause: Throwable?)
            : super(message, cause) {
        this.errorCode = errorCode
    }

    val status: HttpStatus
        get() = errorCode.status
}
